QUAD-108 Created project statistics page and displayed d3 bar graph w…

bharathsharma opened the pull request:
QUAD-108 Created project statistics page and displayed d3 bar graph with dummy data

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
please resolve conflicts

bharathsharma commented:
Merged  and conflicts resolved .

jdamerow commented:
Please resolve conflicts and remove .DS_Store file.

jdamerow commented:
Please remove the .DS_Store file

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -216,4 +216,34 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
+	 * @throws JSONException 
+	 */
+	@RequestMapping(value = "sites/{projectUnixName}/statistics", method = RequestMethod.GET)
+	public String showProjectStatistics(@PathVariable("projectUnixName") String projectUnixName,
```
This method should be in its own controller.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -216,4 +216,34 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
+	@RequestMapping(value = "sites/{projectUnixName}/statistics", method = RequestMethod.GET)
+	public String showProjectStatistics(@PathVariable("projectUnixName") String projectUnixName,
+									   Model model)throws JAXBException, QuadrigaStorageException, JSONException {
+		IProject project = getProjectDetails(projectUnixName);
```
This is just one line of code, don't use an extra method for that.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -216,4 +216,34 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
+		}
+ 
+		String data = "[{\"conceptId\":\"ID-1\",\"description\":\"Desc1\",\"frequency\":\".01\"},{\"conceptId\":\"ID-2\",\"description\":\"Desc2\",\"frequency\":\".03\"},{\"conceptId\":\"ID-3\",\"description\":\"Desc3\",\"frequency\":\".01\"},{\"conceptId\":\"ID-4\",\"description\":\"Desc4\",\"frequency\":\".02\"},{\"conceptId\":\"ID-5\",\"description\":\"Desc5\",\"frequency\":\".04\"}]";
+		System.out.println("Json obj "+data);
```
remove

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -216,4 +216,34 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
+			return "auth/accessissue";
+		}
+ 
+		String data = "[{\"conceptId\":\"ID-1\",\"description\":\"Desc1\",\"frequency\":\".01\"},{\"conceptId\":\"ID-2\",\"description\":\"Desc2\",\"frequency\":\".03\"},{\"conceptId\":\"ID-3\",\"description\":\"Desc3\",\"frequency\":\".01\"},{\"conceptId\":\"ID-4\",\"description\":\"Desc4\",\"frequency\":\".02\"},{\"conceptId\":\"ID-5\",\"description\":\"Desc5\",\"frequency\":\".04\"}]";
```
use a json library to do that.

jdamerow replied:
never mind, disregards last comment.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/projectstatistics.jsp:
```diff
@@ -0,0 +1,34 @@
+	<h2>Project Statistics</h2>
+	<div id='stats'></div>
+</body>
+</html>
```
don't provide a complete HTML page (with html, body, ... tags). The skeleton does that.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp:
```diff
@@ -1,10 +1,6 @@
+<li ${currentPage == "networks" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
+<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
```
what about the search url?

jdamerow commented on Quadriga/src/main/webapp/resources/txt-layout/js/d3projectstatistics.js:
```diff
@@ -0,0 +1,77 @@
+ * 
+ */
+
+function d3projectstatistics(data) {
```
Let's use proper camel case here.

jdamerow replied:
and indentation

jdamerow commented:
See comments. Please reopen PR when you're done.

bharathsharma commented:
All issues addressed.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
Cool. Please resolve conflicts and connect it with the backend.

bharathsharma commented:
Merged with story/QUAD-114 and resolved conflicts
@ajaymodi FYR

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
Still: please remove .DS_Store file.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp:
```diff
@@ -1,10 +1,6 @@
+<li ${currentPage == "home" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
+<li ${currentPage == "networks" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
```
you want the Statistics menu entry to be active when you are on the statistics page, so you should use your own currentPage value.

jdamerow commented:
ok to test

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -0,0 +1,125 @@
+
+    private int getCount(List<IConceptStats> conceptList) {
+
+        int cnt = Integer.parseInt(env.getProperty("project.stats.topcount"));
```
This line raises the following exception:

```
edu.asu.spring.quadriga.exceptions.QuadrigaException: null
    java.lang.Integer.parseInt(Integer.java:542)
    java.lang.Integer.parseInt(Integer.java:615)
    edu.asu.spring.quadriga.web.publicwebsite.ProjectStatsController.getCount(ProjectStatsController.java:51)
```

jdamerow commented:
please reopen PR when you're done

bharathsharma commented:
Resolved all issues

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
retest this please

jdamerow commented:
Great start! A few things and its good to go:
- You need to use the skeleton for public pages (with the blue navigation bar).
- The diagram should be centered.
- the layout of the hover over pop up should be inline with the rest of the layout of the page (no red font, maybe a little smaller font), and the pop ups should be smaller and more squared. Right now it also runs out of the page (see screenshot).

![image](https://cloud.githubusercontent.com/assets/8881141/14195236/cf14d1b4-f76b-11e5-8375-06502e5af212.png)

bharathsharma commented:
Addressed all issues.Attached screen shot for reference.

<img width="1440" alt="screen shot 2016-03-31 at 10 40 20 pm" src="https://cloud.githubusercontent.com/assets/10238564/14198823/3ba16078-f793-11e5-98eb-1c1afc9fd5d4.png">

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
ok to test

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/service/publicwebsite/impl/ProjectStats.java:
```diff
@@ -0,0 +1,76 @@
+ */
+
+@Service
+public class ProjectStats implements IProjectStats {
```
@ajaymodi, please write a test class for this class.

ajaymodi replied:
I have added test class.

jdamerow commented:
Looks great. Is there a reason though, that the diagram is still not centered?

bharathsharma commented:
I was centering the diagram with respect to first bar. Ill fix it in next release

jdamerow commented on Quadriga/src/test/java/edu/asu/spring/quadriga/service/publicwebsite/impl/ProjectStatsTest.java:
```diff
@@ -0,0 +1,87 @@
+        Mockito.when(mockedTransformationManager.getTransformedNetwork("id")).thenReturn(transformedNetwork1);
+        Mockito.when(mockedTransformationManager.getTransformedNetwork("id2")).thenReturn(transformedNetwork2);
+                
+        assertEquals("conceptCount", conceptStatsList, projectStatsUnderTest.getConceptCount(networkList));
```
YOu should also test if the sorting works correctly. Write a separate test method for that and it probably makes sense to put the set up code into a before method.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -0,0 +1,125 @@
+    @RequestMapping(value = "sites/{projectUnixName}/statistics", method = RequestMethod.GET)
+    public String showProjectStatistics(
+            @PathVariable("projectUnixName") String projectUnixName, Model model)
+            throws JAXBException, QuadrigaStorageException {
```
There are conflicts in your code. Please merge them. Also, there is now an aspect that injects the correct project given the unix name. Try to incorporate that:
https://diging.atlassian.net/wiki/display/QUAD/Injecting+projects+by+unix+names

You can then loose the next line.

jdamerow commented:
please resolve conflicts and incorporate comments. Reopen PR when you're done.
